tests/gcp/test_fg.py

```diff
@@ -51,3 +56,72 @@ def no_op(data, model):
     )
     assert isinstance(f, float)
     assert all(isinstance(g_i, np.ndarray) for g_i in g)
+
+
+def test_evaluate_sptensor():
+    model = ttb.import_data("tests/data/ktensor_low_rank_small.tns")
+    data = ttb.import_data("tests/data/sptensor_low_rank_small.tns")
+    EPS = 1e-01
+
+    # Computed with MATLAB TTB v3.5
+    function_values = [
+        5.114502408482430,
+        -61.434016498399011,
+        -125.9778934454338,
+        -26.214062308801779,
+        9751.193889205471,
+        -7705.253241233131,
+        -3844.118857614557,
+        1.659013125911521,
+        18.038214072232726,
+        438.0174726326384,
+    ]
```

Comment:
Small preference would be if we had a way to independently generate these dynamically in a way independent of the source code that would be nice. For example, if MATLAB changes we won't know these went stale or how to generate them.

If that's not feasible at the moment then a smaller change that probably makes maintenance easier is making these maps from name to value. That way if the order of `Objectives` happens to need to change or we add a new objective MATLAB doesn't have it doesn't automatically break this test.